src/duqtools/submit.py

```diff
@@ -140,12 +140,19 @@ def submit(*, force: bool, max_jobs: int, schedule: bool, array: bool,
 
     debug('Submit config: %s', cfg.submit)
 
-    workspace = WorkDirectory.parse_obj(cfg.workspace)
+    workspace = WorkDirectory()
     runs = workspace.runs
+    run_dict = {run.shortname: run for run in WorkDirectory().runs}
 
     if len(resubmit) > 0:
+        jobs = []
         force = True
-        jobs = [Job(Path(run)) for run in resubmit]
+        for run in resubmit:
+            if run in run_dict:
+                jobs.append(Job(run_dict[run].dirname))
+            else:
+                jobs.append(Job(Path(run)))
+
     else:
         jobs = [Job(run.dirname) for run in runs]
```

Comment:
This whole section is a bit of a mess to me. It seems like it logically mixes two workflows, I'm trying to make sense of what is happening here.

The first part tries to handle resubmissions, the second part the 'normal' submit workflow.

How about something like this:

```python
    if resubmit:
        force = True
        
        run_dict = {run.shortname: run.dirname for run in WorkDirectory().runs}

        jobs = [Job(run_dict.get(run, Path(run)) for run in resubmit]
    else:
        jobs = [Job(run.dirname) for run in WorkDirectory().runs]
```

Reply:
created helper function, should be very clear now